openfeature/exposure.go

```diff
@@ -0,0 +1,262 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package openfeature
+
+import (
+	"bytes"
+	"cmp"
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/DataDog/dd-trace-go/v2/internal"
+	"github.com/DataDog/dd-trace-go/v2/internal/env"
+	"github.com/DataDog/dd-trace-go/v2/internal/globalconfig"
+	"github.com/DataDog/dd-trace-go/v2/internal/log"
+	telemetrylog "github.com/DataDog/dd-trace-go/v2/internal/telemetry/log"
+	jsoniter "github.com/json-iterator/go"
+)
+
+const (
+	// defaultExposureFlushInterval is the default interval for flushing exposure events
+	// Matches the dd-trace-js implementation (1 second)
+	defaultExposureFlushInterval = 1 * time.Second
+
+	// exposureEndpoint is the EVP proxy endpoint for exposure events
+	exposureEndpoint = "/evp_proxy/v2/api/v2/exposures"
+
+	// evpSubdomainHeader is the HTTP header name for EVP subdomain routing
+	evpSubdomainHeader = "X-Datadog-EVP-Subdomain"
+
+	// evpSubdomainValue is the subdomain value for event platform intake
+	evpSubdomainValue = "event-platform-intake"
+
+	// defaultHTTPTimeout is the timeout for HTTP requests to the agent
+	defaultHTTPTimeout = 5 * time.Second
+)
+
+// exposureEvent represents a single feature flag evaluation exposure event.
+// It matches the schema defined in exposure.json.
+type exposureEvent struct {
+	Timestamp  int64              `json:"timestamp"`
+	Allocation exposureAllocation `json:"allocation"`
+	Flag       exposureFlag       `json:"flag"`
+	Variant    exposureVariant    `json:"variant"`
+	Subject    exposureSubject    `json:"subject"`
+}
+
+// exposureAllocation represents allocation information in an exposure event
+type exposureAllocation struct {
+	Key string `json:"key"`
+}
+
+// exposureFlag represents flag information in an exposure event
+type exposureFlag struct {
+	Key string `json:"key"`
+}
+
+// exposureVariant represents variant information in an exposure event
+type exposureVariant struct {
+	Key string `json:"key"`
+}
```

Comment:
Could those 3 structs be merged into a single `exposureKey` ? 

```suggestion
type exposureKey struct {
	Key string `json:"key"`
}
```